MAINT: Update Python version support documentation (#2740)
py-pdf/pypdf: docs/user/installation.md

## Python Version Support

Since pypdf 4.0, every release, including point releases, should work with all
supported versions of [Python](https://devguide.python.org/versions/). Thus

> every point release is designed to work with all concurrent Python versions,

Reviewer: ```suggestion
every point release is designed to work with the current Python versions,
```
